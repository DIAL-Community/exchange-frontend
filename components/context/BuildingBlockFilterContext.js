import React, { createContext, useState } from 'react'

const BuildingBlockFilterContext = createContext()
const BuildingBlockFilterDispatchContext = createContext()

const BuildingBlockFilterProvider = ({ children }) => {
  const [showMature, setShowMature] = useState(false)
  const [sdgs, setSDGs] = useState([])
  const [useCases, setUseCases] = useState([])
  const [workflows, setWorkflows] = useState([])
  const [categoryTypes, setCategoryTypes] = useState([])

  const [search, setSearch] = useState('')
  const [sortColumn, setSortColumn] = useState('name')
  const [sortDirection, setSortDirection] = useState('asc')

  const setSdgs = setSDGs

  const buildingBlockFilterValues = {
    showMature, sdgs, useCases, workflows, categoryTypes, search, sortColumn, sortDirection
  }
  const buildingBlockFilterDispatchValues = {
    setShowMature,
    setSDGs,
    setSdgs,
    setUseCases,
    setWorkflows,
    setCategoryTypes,
    setSearch,
    setSortColumn,
    setSortDirection
  }

  return (
    <BuildingBlockFilterContext.Provider value={buildingBlockFilterValues}>
      <BuildingBlockFilterDispatchContext.Provider value={buildingBlockFilterDispatchValues}>
        {children}
      </BuildingBlockFilterDispatchContext.Provider>
    </BuildingBlockFilterContext.Provider>
  )
}

export { BuildingBlockFilterProvider, BuildingBlockFilterContext, BuildingBlockFilterDispatchContext }
