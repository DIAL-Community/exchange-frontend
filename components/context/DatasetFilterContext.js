import { createContext, useState } from 'react'

const DatasetFilterContext = createContext()
const DatasetFilterDispatchContext = createContext()

const DatasetFilterProvider = ({ children }) => {
  const [origins, setOrigins] = useState([])
  const [countries, setCountries] = useState([])
  const [sectors, setSectors] = useState([])
  const [organizations, setOrganizations] = useState([])
  const [sdgs, setSdgs] = useState([])
  const [tags, setTags] = useState([])
  const [datasetTypes, setDatasetTypes] = useState([])

  const [search, setSearch] = useState('')

  const [pageNumber, setPageNumber] = useState(0)
  const [pageOffset, setPageOffset] = useState(0)

  const datasetFilterValues = {
    origins,
    countries,
    sectors,
    organizations,
    sdgs,
    tags,
    datasetTypes,
    search,
    pageOffset,
    pageNumber
  }
  const datasetFilterDispatchValues = {
    setOrigins,
    setCountries,
    setSectors,
    setOrganizations,
    setSdgs,
    setTags,
    setDatasetTypes,
    setSearch,
    setPageNumber,
    setPageOffset
  }

  return (
    <DatasetFilterContext.Provider value={datasetFilterValues}>
      <DatasetFilterDispatchContext.Provider value={datasetFilterDispatchValues}>
        {children}
      </DatasetFilterDispatchContext.Provider>
    </DatasetFilterContext.Provider>
  )
}

export { DatasetFilterProvider, DatasetFilterContext, DatasetFilterDispatchContext }
