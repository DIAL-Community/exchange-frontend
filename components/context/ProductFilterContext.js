import { createContext, useState } from 'react'

const ProductFilterContext = createContext()
const ProductFilterDispatchContext = createContext()

const ProductFilterProvider = ({ children }) => {
  const [isEndorsed, setIsEndorsed] = useState(false)
  const [productDeployable, setProductDeployable] = useState(false)
  const [origins, setOrigins] = useState([])
  const [countries, setCountries] = useState([])
  const [sectors, setSectors] = useState([])
  const [organizations, setOrganizations] = useState([])
  const [sdgs, setSdgs] = useState([])
  const [tags, setTags] = useState([])
  const [useCases, setUseCases] = useState([])
  const [workflows, setWorkflows] = useState([])
  const [buildingBlocks, setBuildingBlocks] = useState([])
  const [endorsers, setEndorsers] = useState([])
  const [licenseTypes, setLicenseTypes] = useState([])
  const [isLinkedWithDpi, setIsLinkedWithDpi] = useState(false)
  const [showGovStackOnly, setShowGovStackOnly] = useState(false)
  const [showDpgaOnly, setShowDpgaOnly] = useState(false)

  const [softwareCategories, setSoftwareCategories] = useState([])
  const [softwareFeatures, setSoftwareFeatures] = useState([])
  const [comparedProducts, setComparedProducts] = useState([])

  const [search, setSearch] = useState('')

  const [pageNumber, setPageNumber] = useState(0)
  const [pageOffset, setPageOffset] = useState(0)

  const productFilterValues = {
    isEndorsed,
    productDeployable,
    origins,
    countries,
    sectors,
    organizations,
    sdgs,
    tags,
    useCases,
    workflows,
    buildingBlocks,
    endorsers,
    licenseTypes,
    isLinkedWithDpi,
    showGovStackOnly,
    showDpgaOnly,
    softwareCategories,
    softwareFeatures,
    comparedProducts,
    search,
    pageOffset,
    pageNumber
  }

  const productFilterDispatchValues = {
    setIsEndorsed,
    setProductDeployable,
    setOrigins,
    setCountries,
    setSectors,
    setOrganizations,
    setSdgs,
    setTags,
    setUseCases,
    setWorkflows,
    setBuildingBlocks,
    setEndorsers,
    setLicenseTypes,
    setIsLinkedWithDpi,
    setShowGovStackOnly,
    setShowDpgaOnly,
    setSoftwareCategories,
    setSoftwareFeatures,
    setComparedProducts,
    setSearch,
    setPageNumber,
    setPageOffset
  }

  return (
    <ProductFilterContext.Provider value={productFilterValues}>
      <ProductFilterDispatchContext.Provider value={productFilterDispatchValues}>
        {children}
      </ProductFilterDispatchContext.Provider>
    </ProductFilterContext.Provider>
  )
}

export { ProductFilterProvider, ProductFilterContext, ProductFilterDispatchContext }
