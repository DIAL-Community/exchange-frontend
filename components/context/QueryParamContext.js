import { createContext, useState } from 'react'

const QueryParamContext = createContext()
const QueryParamContextProvider = ({ children }) => {
  const [interactionDetected, setInteractionDetected] = useState(false)
  return (
    <QueryParamContext.Provider value={{ ...{ interactionDetected, setInteractionDetected } }}>
      {children}
    </QueryParamContext.Provider>
  )
}

export { QueryParamContextProvider, QueryParamContext }
