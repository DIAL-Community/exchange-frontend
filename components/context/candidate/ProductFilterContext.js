import React, { createContext, useState } from 'react'

const ProductFilterContext = createContext()
const ProductFilterDispatchContext = createContext()

const ProductFilterProvider = ({ children }) => {
  const [search, setSearch] = useState('')

  const candidateProductFilterValues = {
    search
  }
  const candidateProductFilterDispatchValues = {
    setSearch
  }

  return (
    <ProductFilterContext.Provider value={candidateProductFilterValues}>
      <ProductFilterDispatchContext.Provider value={candidateProductFilterDispatchValues}>
        {children}
      </ProductFilterDispatchContext.Provider>
    </ProductFilterContext.Provider>
  )
}

export { ProductFilterProvider, ProductFilterContext, ProductFilterDispatchContext }
