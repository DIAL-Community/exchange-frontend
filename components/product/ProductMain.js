import ProductMainLeft from './ProductMainLeft'
import ProductMainRight from './ProductMainRight'

const ProductMain = ({ activeTab }) => {
  return (
    <div className='px-4 lg:px-8 xl:px-56'>
      <div className='grid grid-cols-4 gap-x-8'>
        <div className='hidden md:block col-span-1'>
          <ProductMainLeft activeTab={activeTab} />
        </div>
        <div className='col-span-4 md:col-span-3'>
          <ProductMainRight activeTab={activeTab} />
        </div>
      </div>
    </div>
  )
}

export default ProductMain
