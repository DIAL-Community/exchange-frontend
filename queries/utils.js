const MIN_SEARCH_VALUE_CHARACTERS = 2

export const fetchSelectOptions = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}

export const fetchSelectOptionsWithMature = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input,
    mature: true
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}
