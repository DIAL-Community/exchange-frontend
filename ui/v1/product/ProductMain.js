import ProductMainLeft from './ProductMainLeft'
import ProductMainRight from './ProductMainRight'

const ProductMain = ({ activeTab }) => {
  return (
    <div className='px-4 lg:px-8 xl:px-56'>
      <div className='grid grid-cols-3 gap-x-8'>
        <div className='hidden md:block col-span-1'>
          <ProductMainLeft activeTab={activeTab} />
        </div>
        <div className='col-span-3 md:col-span-2'>
          <ProductMainRight activeTab={activeTab} />
        </div>
      </div>
    </div>
  )
}

export default ProductMain
