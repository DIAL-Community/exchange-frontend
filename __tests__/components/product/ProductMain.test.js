import { screen } from '@testing-library/dom'
import { render } from '../../test-utils'
import { mockNextUseRouter } from '../../utils/nextMockImplementation'
import ProductMain from '../../../components/product/ProductMain'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import {
  PRODUCT_PAGINATION_ATTRIBUTES_QUERY,
  PAGINATED_PRODUCTS_QUERY
} from '../../../components/shared/query/product'
import { QueryParamContextProvider } from '../../../components/context/QueryParamContext'
import { ProductFilterProvider } from '../../../components/context/ProductFilterContext'
import { paginatedProducts, productPaginationAttribute } from './data/ProductMain.data'

mockNextUseRouter()
describe('Unit tests for the product main page.', () => {
  test('Should render list of products.', async () => {
    const mockProductPaginationAttribute = generateMockApolloData(
      PRODUCT_PAGINATION_ATTRIBUTES_QUERY,
      {
        search: '',
        useCases: [],
        buildingBlocks: [],
        sectors: [],
        tags: [],
        licenseTypes: [],
        sdgs: [],
        workflows: [],
        origins: [],
        isLinkedWithDpi: false
      },
      null,
      productPaginationAttribute
    )
    const mockPaginatedProducts = generateMockApolloData(
      PAGINATED_PRODUCTS_QUERY,
      {
        search: '',
        useCases: [],
        buildingBlocks: [],
        sectors: [],
        tags: [],
        licenseTypes: [],
        sdgs: [],
        workflows: [],
        origins: [],
        isLinkedWithDpi: false,
        limit: 8,
        offset: 0
      },
      null,
      paginatedProducts
    )

    const { container } = render(
      <CustomMockedProvider mocks={[mockPaginatedProducts, mockProductPaginationAttribute]}>
        <QueryParamContextProvider>
          <ProductFilterProvider>
            <ProductMain activeTab={0} />
          </ProductFilterProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('AP Latam')).toBeInTheDocument()
    expect(await screen.findByText('AIDR')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
