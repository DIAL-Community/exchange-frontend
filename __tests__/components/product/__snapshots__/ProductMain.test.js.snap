// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the product main page. Should render list of products. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="px-4 lg:px-8 xl:px-56"
  >
    <div
      class="grid grid-cols-3 gap-x-8"
    >
      <div
        class="hidden md:block col-span-1"
      >
        <div
          class="bg-dial-slate-100 h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="flex flex-col gap-y-2"
              >
                <div
                  class="text-sm font-semibold text-dial-sapphire"
                >
                  Filter your results by
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <div
                  class="flex flex-col gap-y-3"
                >
                  <a
                    class="flex"
                    href="#"
                  >
                    <div
                      class="text-dial-stratos text-sm py-2"
                    >
                      Use Case
                    </div>
                    <svg
                      class="ml-auto text-dial-stratos my-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
                      />
                    </svg>
                  </a>
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <div
                  class="flex flex-col gap-y-3"
                >
                  <a
                    class="flex"
                    href="#"
                  >
                    <div
                      class="text-dial-stratos text-sm py-2"
                    >
                      Building Block
                    </div>
                    <svg
                      class="ml-auto text-dial-stratos my-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
                      />
                    </svg>
                  </a>
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <div
                  class="flex flex-col gap-y-3"
                >
                  <a
                    class="flex"
                    href="#"
                  >
                    <div
                      class="text-dial-stratos text-sm py-2"
                    >
                      Sector
                    </div>
                    <svg
                      class="ml-auto text-dial-stratos my-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
                      />
                    </svg>
                  </a>
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <div
                  class="flex flex-col gap-y-3"
                >
                  <a
                    class="flex"
                    href="#"
                  >
                    <div
                      class="text-dial-stratos text-sm py-2"
                    >
                      Tag
                    </div>
                    <svg
                      class="ml-auto text-dial-stratos my-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
                      />
                    </svg>
                  </a>
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <label
                  class="flex py-2"
                >
                  <input
                    class="h-5 w-5"
                    type="checkbox"
                  />
                  <span
                    class="mx-2 my-auto text-sm"
                  >
                    Show only linked with DPI
                  </span>
                </label>
                <hr
                  class="border-b border-dial-slate-200"
                />
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <div
                  class="text-sm font-semibold text-dial-sapphire"
                >
                  <button
                    type="button"
                  >
                    <div
                      class="flex text-dial-sapphire gap-3"
                    >
                      <div
                        class="my-auto"
                      >
                        Refine with more filters
                      </div>
                      <div
                        class="ml-auto py-2 text-xl"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 448 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M201.4 342.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 274.7 86.6 137.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                          />
                        </svg>
                      </div>
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="flex flex-col gap-3 py-3"
            >
              <div
                class="text-dial-sapphire font-semibold"
              >
                Bookmark
              </div>
              <button
                class="group flex flex-row gap-x-3"
                type="button"
              >
                <img
                  alt="Logo of Bookmark."
                  class="object-contain"
                  height="40"
                  src="/ui/v1/bookmark-icon.svg"
                  width="40"
                />
                <div
                  class="my-auto border-b border-transparent group-hover:border-dial-slate-500"
                >
                  <div
                    class="text-sm text-dial-stratos text-left"
                  >
                    Bookmark this page.
                  </div>
                </div>
              </button>
              <div
                class="text-sm text-dial-stratos"
              >
                Your bookmarks can be found in your Account page.
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
            <div
              class="flex flex-col gap-3 py-3"
            >
              <div
                class="text-dial-sapphire font-semibold"
              >
                Share
              </div>
              <div
                class="flex flex-row gap-3"
              >
                <img
                  alt="Logo of LinkedIn."
                  class="object-contain"
                  height="40"
                  src="/ui/v1/linkedin-icon.svg"
                  width="40"
                />
                <img
                  alt="Logo of Email."
                  class="object-contain"
                  height="40"
                  src="/ui/v1/email-icon.svg"
                  width="40"
                />
                <img
                  alt="Logo of Copy."
                  class="object-contain"
                  height="40"
                  src="/ui/v1/copy-icon.svg"
                  width="40"
                />
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
          </div>
        </div>
      </div>
      <div
        class="col-span-3 md:col-span-2"
      >
        <div
          class="min-h-[50vh]"
        >
          <div
            class="py-3"
          >
            <div
              class="flex flex-row gap-3"
            >
              <div
                class="block md:hidden relative"
              >
                <button
                  class="my-auto h-full"
                  type="button"
                >
                  <svg
                    class="text-2xl text-dial-meadow mx-auto"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 416c0 17.7 14.3 32 32 32l54.7 0c12.3 28.3 40.5 48 73.3 48s61-19.7 73.3-48L480 448c17.7 0 32-14.3 32-32s-14.3-32-32-32l-246.7 0c-12.3-28.3-40.5-48-73.3-48s-61 19.7-73.3 48L32 384c-17.7 0-32 14.3-32 32zm128 0a32 32 0 1 1 64 0 32 32 0 1 1 -64 0zM320 256a32 32 0 1 1 64 0 32 32 0 1 1 -64 0zm32-80c-32.8 0-61 19.7-73.3 48L32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l246.7 0c12.3 28.3 40.5 48 73.3 48s61-19.7 73.3-48l54.7 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-54.7 0c-12.3-28.3-40.5-48-73.3-48zM192 128a32 32 0 1 1 0-64 32 32 0 1 1 0 64zm73.3-64C253 35.7 224.8 16 192 16s-61 19.7-73.3 48L32 64C14.3 64 0 78.3 0 96s14.3 32 32 32l86.7 0c12.3 28.3 40.5 48 73.3 48s61-19.7 73.3-48L480 128c17.7 0 32-14.3 32-32s-14.3-32-32-32L265.3 64z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="ml-auto w-full"
              >
                <label
                  class="relative"
                  for="search-bar"
                >
                  <input
                    class="w-full border text-sm"
                    id="search-bar"
                    placeholder="Search"
                    type="text"
                    value=""
                  />
                  <svg
                    class="absolute top-1/2 transform -translate-y-1/2 right-2"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </label>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-3"
          >
            <div>
              <div
                class="relative"
              >
                <a
                  href="/products/aidr"
                >
                  <div
                    class="px-4 py-6 rounded-lg min-h-[13.5rem] bg-dial-spearmint"
                  >
                    <div
                      class="flex flex-col lg:flex-row gap-x-6 gap-y-3"
                    >
                      <div
                        class="w-20 h-20 mx-auto bg-white border"
                      >
                        <img
                          alt="Logo of Product."
                          class="object-contain w-16 h-16 mx-auto my-2"
                          src="http://localhost:3000//assets/products/aidr.png"
                        />
                      </div>
                      <div
                        class="flex flex-col gap-y-3 max-w-3xl lg:w-10/12"
                      >
                        <div
                          class="text-lg font-semibold text-dial-meadow"
                        >
                          AIDR
                        </div>
                        <div
                          class="line-clamp-4 text-dial-stratos"
                        />
                        <div
                          class="flex gap-x-2 text-dial-stratos"
                        >
                          <div
                            class="text-sm"
                          >
                            SDGs
                             (
                            0
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Building Blocks
                             (
                            0
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Countries
                             (
                            0
                            )
                          </div>
                        </div>
                        <div
                          class="flex gap-2 text-xs text-white"
                        />
                      </div>
                    </div>
                  </div>
                </a>
                <div
                  class="absolute top-2 right-2"
                >
                  <label
                    class="ml-auto flex gap-x-2 text-sm"
                  >
                    <input
                      class="ring-0 focus:ring-0 h-5 w-5"
                      type="checkbox"
                    />
                    Compare
                  </label>
                </div>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/products/ap_latam"
                >
                  <div
                    class="px-4 py-6 rounded-lg min-h-[13.5rem] false"
                  >
                    <div
                      class="flex flex-col lg:flex-row gap-x-6 gap-y-3"
                    >
                      <div
                        class="w-20 h-20 mx-auto"
                      >
                        <img
                          alt="Logo of Product."
                          class="object-contain w-16 h-16"
                          src="http://localhost:3000//assets/products/product_placeholder.svg"
                        />
                      </div>
                      <div
                        class="flex flex-col gap-y-3 max-w-3xl lg:w-10/12"
                      >
                        <div
                          class="text-lg font-semibold text-dial-meadow"
                        >
                          AP Latam
                        </div>
                        <div
                          class="line-clamp-4 text-dial-stratos"
                        >
                          Some parsed description.
                        </div>
                        <div
                          class="flex gap-x-2 text-dial-stratos"
                        >
                          <div
                            class="text-sm"
                          >
                            SDGs
                             (
                            4
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Building Blocks
                             (
                            0
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Countries
                             (
                            0
                            )
                          </div>
                        </div>
                        <div
                          class="flex gap-2 text-xs text-white"
                        >
                          <div
                            class="border border-dial-iris-blue text-dial-iris-blue rounded px-6 py-1"
                          >
                            GOVSTACK
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
                <div
                  class="absolute top-2 right-2"
                >
                  <label
                    class="ml-auto flex gap-x-2 text-sm"
                  >
                    <input
                      class="ring-0 focus:ring-0 h-5 w-5"
                      type="checkbox"
                    />
                    Compare
                  </label>
                </div>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/products/aajeevika"
                >
                  <div
                    class="px-4 py-6 rounded-lg min-h-[13.5rem] bg-dial-spearmint"
                  >
                    <div
                      class="flex flex-col lg:flex-row gap-x-6 gap-y-3"
                    >
                      <div
                        class="w-20 h-20 mx-auto bg-white border"
                      >
                        <img
                          alt="Logo of Product."
                          class="object-contain w-16 h-16 mx-auto my-2"
                          src="http://localhost:3000//assets/products/aajeevika.png"
                        />
                      </div>
                      <div
                        class="flex flex-col gap-y-3 max-w-3xl lg:w-10/12"
                      >
                        <div
                          class="text-lg font-semibold text-dial-meadow"
                        >
                          Aajeevika
                        </div>
                        <div
                          class="line-clamp-4 text-dial-stratos"
                        >
                          Some parsed description.
                        </div>
                        <div
                          class="flex gap-x-2 text-dial-stratos"
                        >
                          <div
                            class="text-sm"
                          >
                            SDGs
                             (
                            2
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Building Blocks
                             (
                            0
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Countries
                             (
                            0
                            )
                          </div>
                        </div>
                        <div
                          class="flex gap-2 text-xs text-white"
                        />
                      </div>
                    </div>
                  </div>
                </a>
                <div
                  class="absolute top-2 right-2"
                >
                  <label
                    class="ml-auto flex gap-x-2 text-sm"
                  >
                    <input
                      class="ring-0 focus:ring-0 h-5 w-5"
                      type="checkbox"
                    />
                    Compare
                  </label>
                </div>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/products/accessmod"
                >
                  <div
                    class="px-4 py-6 rounded-lg min-h-[13.5rem] false"
                  >
                    <div
                      class="flex flex-col lg:flex-row gap-x-6 gap-y-3"
                    >
                      <div
                        class="w-20 h-20 mx-auto bg-white border"
                      >
                        <img
                          alt="Logo of Product."
                          class="object-contain w-16 h-16 mx-auto my-2"
                          src="http://localhost:3000//assets/products/accessmod.png"
                        />
                      </div>
                      <div
                        class="flex flex-col gap-y-3 max-w-3xl lg:w-10/12"
                      >
                        <div
                          class="text-lg font-semibold text-dial-meadow"
                        >
                          AccessMod
                        </div>
                        <div
                          class="line-clamp-4 text-dial-stratos"
                        >
                          Some parsed description.
                        </div>
                        <div
                          class="flex gap-x-2 text-dial-stratos"
                        >
                          <div
                            class="text-sm"
                          >
                            SDGs
                             (
                            1
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Building Blocks
                             (
                            0
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Countries
                             (
                            0
                            )
                          </div>
                        </div>
                        <div
                          class="flex gap-2 text-xs text-white"
                        >
                          <div
                            class="border border-dial-iris-blue text-dial-iris-blue rounded px-6 py-1"
                          >
                            GOVSTACK
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
                <div
                  class="absolute top-2 right-2"
                >
                  <label
                    class="ml-auto flex gap-x-2 text-sm"
                  >
                    <input
                      class="ring-0 focus:ring-0 h-5 w-5"
                      type="checkbox"
                    />
                    Compare
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-y-4 my-6"
          >
            <hr
              class="border-b border-dial-slate-300"
            />
            <div
              class="flex flex-col xl:flex-row gap-x-16 gap-y-3"
            >
              <div
                class="my-auto text-sm font-semibold"
              >
                Showing 1 - 8 of 23 total
              </div>
              <ul
                aria-label="Pagination"
                class="flex gap-x-3 text-sm text-dial-slate-300 font-semibold"
                role="navigation"
              >
                <li
                  class="border border-current rounded-md opacity-30"
                >
                  <a
                    aria-disabled="true"
                    aria-label="Previous page"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500 "
                    rel="prev"
                    role="button"
                    tabindex="-1"
                  >
                    &lt;
                  </a>
                </li>
                <li
                  class="border border-current rounded-md text-dial-iris-blue"
                >
                  <a
                    aria-current="page"
                    aria-label="Page 1 is your current page"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    rel="canonical"
                    role="button"
                    tabindex="-1"
                  >
                    1
                  </a>
                </li>
                <li
                  class="border border-current rounded-md"
                >
                  <a
                    aria-label="Page 2"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    rel="next"
                    role="button"
                    tabindex="0"
                  >
                    2
                  </a>
                </li>
                <li
                  class="border border-current rounded-md"
                >
                  <a
                    aria-label="Page 3"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    role="button"
                    tabindex="0"
                  >
                    3
                  </a>
                </li>
                <li
                  class="border border-current rounded-md"
                >
                  <a
                    aria-disabled="false"
                    aria-label="Next page"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    rel="next"
                    role="button"
                    tabindex="0"
                  >
                    &gt;
                  </a>
                </li>
              </ul>
            </div>
            <hr
              class="border-b border-dial-slate-300"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
